File: src/main/kotlin/adaptor_pattern/java_case/PrintBannerDelegate.java
package adaptor_pattern.java_case;

public class PrintBannerDelegate extends PrintDelegate {
    private Banner banner;

    public PrintBannerDelegate(String string) {
        this.banner = new Banner(string);
    }

    @Override
    public void printWeak() {
        banner.showWithParen();
    }

    @Override
    public void printStrong() {
        banner.showWithAster();
    }
}

File: src/main/kotlin/adaptor_pattern/kotlin_case/PrintBannerDelegate.kt
package adaptor_pattern.kotlin_case

class PrintBannerDelegate(private val string:String):PrintDelegate() {
    private val banner = Banner(string)

    override fun printWeak() {
        banner.showWithParen()
    }

    override fun printStrong() {
        banner.showWithAster()
    }
}